client/hotrod/src/main/java/org/infinispan/hotrod/impl/Util.java

```diff
@@ -77,9 +77,9 @@ public static <T> T await(CompletableFuture<T> cf, long timeoutMillis) {
 
    protected static RuntimeException rewrap(ExecutionException e) {
       if (e.getCause() instanceof HotRodClientException) {
-         return (HotRodClientException) e.getCause();
+         return new HotRodClientException(e);
```

Comment:
I think we'll need to copy the original message to fix the tests:

```suggestion
         HotRodClientException hrce = (HotRodClientException) e.getCause();
         return new HotRodClientException(hrce.getMessage(), hrce);
```

Maybe we'll need another branch here for another type of exception, as the cause is one step deeper.

Reply:
I am also not the biggest fan of having more nested exceptions. This will create a
HRCException which is caused by an ExecutionException wich is caused by a HRCException for every client exception, which seems extremely excessive.

I haven't totally thought it through but it might be worth to use Throwable.setStackTrace instead. The end user will not care about any of the netty stack frames as it just comes from parsing the response from the server. We could create a HRCException that has the message of the cause but the frames of the ExcecutionException.

Reply:
+1 for using the Throwable.setStackTrace. We can add `e.getCause().setStackTrace(e.getStackTrace());` before throwing and should work as expected.